### PR TITLE
Add support for distribution_scope in patch request

### PR DESCRIPTION
**Pull request by yashvardhannanavati**
Refers to CLOUDDST-4297

### CODE REVIEW

**Comment by yashvardhannanavati** [comment]
@pbortlov Added the change to upgrade cryptography as a separate commit:)

File: tests/test_web/test_api_v1.py [hunk 1]
```diff
@@ -902,6 +902,7 @@ def test_patch_request_add_success(mock_smfsc, db, minimal_request_add, worker_a
         'state_reason': 'All done!',
         'index_image': 'index:image',
         'binary_image_resolved': 'binary-image@sha256:1234',
+        'distribution_scope': 'stage',
```
Comment (lipoja) [inline tests/test_web/test_api_v1.py:905]: `distribution_scope` is optional, right? Should not we have test for request when it is not set?

File: iib/web/api_v1.py [hunk 2]
```diff
@@ -433,6 +433,9 @@ def patch_request(request_id):
             bundle_img = Image.get_or_create(bundle)
             bundle_img.operator = operator_img
 
+    if 'distribution_scope' in payload:
+        setattr(request, 'distribution_scope', payload['distribution_scope'])
```
Comment (mprahl) [inline iib/web/api_v1.py:437]: Why are you using `setattr`?
```suggestion
        request.distribution_scope = payload['distribution_scope']
```

File: iib/web/api_v1.py [hunk 3]
```diff
@@ -433,6 +433,9 @@ def patch_request(request_id):
             bundle_img = Image.get_or_create(bundle)
             bundle_img.operator = operator_img
 
+    if 'distribution_scope' in payload:
```
Comment (mprahl) [inline iib/web/api_v1.py:436]: Don't you need to validate the value?

Comment (yashvardhannanavati) [reply iib/web/api_v1.py:436]: We already validate it in build.py. So I don't think we need to do it here as well.

Comment (mprahl) [reply iib/web/api_v1.py:436]: Sure, there is a precedent to validate all values on this API endpoint though. The main reason is that the API is separate from the workers, and trust should be limited.

**Comment by midnightercz** [comment]
Overall looks good

File: iib/web/api_v1.py (hunk 2 above)
Comment (yashvardhannanavati) [reply iib/web/api_v1.py:437]: you're right

File: iib/web/api_v1.py (hunk 3 above)
Comment (yashvardhannanavati) [reply iib/web/api_v1.py:436]: So L377 should throw an error if it's a non-empty string. Are you suggesting to validate something else too?

File: tests/test_web/test_api_v1.py (hunk 1 above)
Comment (yashvardhannanavati) [reply tests/test_web/test_api_v1.py]: good point. added! :)

**Review by mprahl — APPROVED** [review]

**Pull request merged**